docs/USAGE-GUIDE.md

```diff
@@ -493,6 +493,18 @@ int s2n_config_free(struct s2n_config *config);
 
 **s2n_config_free** frees the memory associated with an **s2n_config** object.
 
+### s2n\_config\_set\_security\_policy
+
+```c
+int s2n_config_set_versioned_security_policy(struct s2n_config *config,
+                                      const char *version);
+```
+
+**s2n_config_set_versioned_security_policy** S2N makes opinionated decisions about which ciphers,
+key encapsualtion mechanisms, signature algorithms, and curves can be used together. These
+combinations are called *security policies*. This function is simply a wrapper for
+*s2n_config_set_cipher_preferences*. The name better describes the behavior behind the scenes.
```

Comment:
Should we mark previous ones as deprecate in our docs?

Reply:
We intentionally kept the previous APIs as it would break current customers. We cannot deprecate due to backward compatibility reasons.

Reply:
We won't remove the APIs but I think we should let customers know that there's a better alternative. Those functions should be soft-deprecated and marked with [`__attribute__(deprecated)`](https://gcc.gnu.org/onlinedocs/gcc-4.7.0/gcc/Function-Attributes.html#index-g_t_0040code_007bdeprecated_007d-attribute_002e-2502)

Reply:
The API changes would look something like:

```
#define S2N_DEPRECATED __attribute__((deprecated))

/* Old functions using char* versioning are deprecated (though preserved for backwards compatibility) */
S2N_DEPRECATED
int s2n_config_set_cipher_preferences(struct s2n_config *config, const char *version);

S2N_DEPRECATED
int s2n_connection_set_cipher_preferences(struct s2n_config *config, const char *version);

/* New functions use pointers to the policy structures.            */
/* This removes strcasecmp latency when locating versions. */
S2N_RESULT s2n_config_set_security_policy(struct s2n_config *, struct s2n_security_policy *);
S2N_RESULT s2n_connection_set_security_policy(struct s2n_connection *, struct s2n_security_policy *);
```

This does give a lot of warning spam when building due to the tests and the functions wrapping set_cipher_preferences. I'm looking at ways to overcome that.